### ISSUE TITLE
Promise should return full error object

### ISSUE THREAD

**Issue by ankurk91** [spatie/form-backend-validation#25]
Hi,
Currently this package return `error.response` when promise fails
https://github.com/spatie/form-backend-validation/blob/master/src/Form.js#L173

But i want leverage axios error object, see
https://github.com/mzabriskie/axios#handling-errors

We should return full error object so the developer can fully utilize it.
```js
.catch((error) => {
                    this.processing = false;
                    this.onFail(error.response.data);
                    // return error not just response
                    reject(error);
 });
```

**Comment by freekmurze** [comment]
You're right that there should be a way to get the full error. Unfortunately changing this now would be a breaking change.

Probably we'll change this is the next major version (unless @sebastiandedeyne knows a non breaking way to do this now)

**Comment by huglester** [comment]
as I understand you are going to release v2 anyway? together with laravel 5.5 support

**Comment by sidouglas** [comment]
If there is a network timeout, that error object won't have a response property, and it's going to crash any return with a type error. I'm running into this now.

`` `
.catch((error) => {
                    this.processing = false;
                    if(error.hasOwnProperty('response')){
                     this.onFail(error.response.data); //the data or undefined.
                    }
                    // return error not just response
                    reject(error);
 });
```

**Comment by huglester** [comment]
Thank you! What is the release date of v2? Thank you

**Comment by sebastiandedeyne** [comment]
The minute Laravel 5.5 is officially released, so some time next week :)
